lib/db_utils.go

```diff
@@ -4844,6 +5013,52 @@ func (txnMeta *LikeTxindexMetadata) GetEncoderType() EncoderType {
 	return EncoderTypeLikeTxindexMetadata
 }
 
+type ReactTxindexMetadata struct {
+	// ReactorPublicKeyBase58Check = TransactorPublicKeyBase58Check
+	IsRemove      bool
+	EmojiReaction rune
+
+	PostHashHex string
+	// PosterPublicKeyBase58Check in AffectedPublicKeys
+}
+
+func (txnMeta *ReactTxindexMetadata) RawEncodeWithoutMetadata(blockHeight uint64, skipMetadata ...bool) []byte {
+	var data []byte
+
+	data = append(data, BoolToByte(txnMeta.IsRemove))
+	data = append(data, []byte(string(txnMeta.EmojiReaction))...)
+	data = append(data, EncodeByteArray([]byte(txnMeta.PostHashHex))...)
+	return data
+}
+
+func (txnMeta *ReactTxindexMetadata) RawDecodeWithoutMetadata(blockHeight uint64, rr *bytes.Reader) error {
+	var err error
+
+	txnMeta.IsRemove, err = ReadBoolByte(rr)
```

Comment:
I usually put isRemove as the last one. I think logically we would expect PostHashHex then EmojiReaction then IsRemove